src/Adapter/Guzzle/HttpClient.php

```diff
@@ -67,7 +68,8 @@ private function http2fallback(RequestInterface $request): RequestInterface
         if (
             'https' !== $request->getUri()->getScheme()
             || !\defined('CURL_VERSION_HTTP2')
-            || !(CURL_VERSION_HTTP2 & curl_version()['features'])
+            || (($curlVersion = curl_version()) == false)
+            || !(CURL_VERSION_HTTP2 & $curlVersion['features'])
```

Comment:
@b-viguier Pinging you on the actual code change just in case. Phpstan was not happy because $curlVersion can return false.
Just to be sure I understood what's going on here: We 're checking if current system has HTTP2 capability. If request is not in https, or if the constant `CURL_VERSION_HTTP2` is not defined, or the `features` key is not available, then we fallback to HTTP1.

Reply:
> or the `features` key is not available

Not exactly, we don't check existence of this key (I think we can assume that `curl_version` always returns it), but if the bitmask stored at key `features` contains (or not) the bit corresponding to the feature `CURL_VERSION_HTTP2` 😄 

So far so good 👍